src/Data/Avro/JSON.hs

```diff
@@ -75,32 +75,50 @@ import           Data.Avro.Schema     (Schema, parseAvroJSON)
 import qualified Data.Avro.Schema     as Schema
 import qualified Data.Avro.Types      as Avro
 
+decodeAvroJSON :: Schema -> Aeson.Value -> Result (Avro.Value Schema.Type)
```

Comment:
`Schema` and `Schema.Type` are the same thing. (Yes, this is a bit confusing in practice.) 

Let's stick with `Schema` for consistency and because it is the better name.

Reply:
Also, it would be nice to have some explicit documentation for this function.